src/python/pants/backend/java/lint/google_java_format/subsystem.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from typing import cast
+
+from pants.jvm.resolve.jvm_tool import JvmToolBase
+from pants.util.docutil import git_url
+
+
+class GoogleJavaFormatSubsystem(JvmToolBase):
+    options_scope = "google-java-format"
+    help = "Google Java Format (https://github.com/google/google-java-format)"
+
+    default_version = "1.13.0"
+    default_artifacts = ["com.google.googlejavaformat:google-java-format:{version}"]
+    default_lockfile_resource = (
+        "pants.backend.java.lint.google_java_format",
+        "google_java_format.default.lockfile.txt",
+    )
+    default_lockfile_path = "src/python/pants/backend/java/lint/google_java_format/google_java_format.default.lockfile.txt"
+    default_lockfile_url = git_url(default_lockfile_path)
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--skip",
+            type=bool,
+            default=False,
+            help=(
+                f"Don't use Google Java Format when running `{register.bootstrap.pants_bin_name} fmt` and "
+                f"`{register.bootstrap.pants_bin_name} lint`"
+            ),
+        )
+        register(
+            "--aosp",
+            type=bool,
+            default=False,
+            help=(
+                "Use AOSP style instead of Google Style (4-space indentation). "
+                '("AOSP" is the Android Open Source Project.)'
+            ),
+        )
```

Comment:
Confirming this doesn't need `--args` and `--config-discovery` or `--config`?

Reply:
There aren't any config files. And too many of the arguments relate to the tool's operating mode so we would not want to expose `--args`.

Reply:
As a follow-up, I may see if we should expose any other options like we did for `--aosp` but shouldn't block this PR.

Reply:
On that note, `--aosp` might make more sense being configured via `--args`. That's how we'd do it for Python formatters

Because this is experimental, you could change that without having to deprecate.